torch/nn/quantized/_reference/modules/utils.py

```diff
@@ -0,0 +1,61 @@
+import torch
+
+def _init_weight_qparams(module, weight_qparams):
+    if weight_qparams is None:
+        weight_qparams = {
+            "qscheme": torch.per_tensor_affine,
+            "dtype": torch.quint8,
+            "scale": 1.0,
+            "zero_point": 0
+        }
+    module.weight_qscheme = weight_qparams["qscheme"]
+    module.weight_dtype = weight_qparams["dtype"]
+    assert module.weight_qscheme in [None, torch.per_tensor_affine, torch.per_channel_affine], \
+    Exception(f"qscheme: {module.weight_qscheme} is not support in reference quantized linear module")
+    if module.weight_qscheme is not None:
+        module.register_buffer("weight_scale", torch.tensor(weight_qparams["scale"]))
+        module.register_buffer("weight_zero_point", torch.tensor(weight_qparams["zero_point"]))
```

Comment:
could we move the buffer registration to the actual modules?  I understand that it might decrease code reuse, but it's not intuitive if buffers are created outside of the module code in a util function.  I think in this case it's better to optimize for readability and the code being obvious.

A motivating example where the current code might be confusing is if a user is looking to see which parameters and buffers are defined on the class.  I'd expect most users to assume that they can find all of the parameters and buffers in the class definitions on the inheritance chain.

Reply:
yeah sure